### PR TITLE
Message with actions

### PR DESCRIPTION
**Pull request by nrostanml**
## Description
Added support for andes message with actions.
MDS: https://mobile.mercadolibre.com/mobile/store/landing/applications/com.mercadolibre.andesui.mds.testapp/builds/2603

## Affected component
AndesMessage, AndesButton (internal - non breaking) 
## Frontify component link
https://company-161429.frontify.com/d/kxHCRixezmfK/n-a#/componentes/message
## Screenshots
andes message with actions

## Checks
Are you sure you followed all those steps? If so, repeat after me "I solemnly swear that ...":
   - [x] I have coded an example inside the Demo App,
   - [x] I have added proper tests,
   - [x] I have modified the Changelog.md file,
   - [ ] I have updated GitHub wiki,
   - [x] I have the explicit approval of one or more members of the UX Team,
   - [x] I have checked that this code is ObjC compliant.
## Change type
This is easy, let us know what this code is about:
   - [ ] This code adds a new component
   - [x] This code includes improvements to an existent component
   - [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift [hunk 1]
```diff
@@ -7,9 +7,20 @@
...
+    static func provideConfig(hierarchy: AndesMessageHierarchy, type: AndesMessageType, title: String?, body: String, isDismissable: Bool, primaryActionText: String?, secondaryActionText: String?) -> AndesMessageViewConfig {
         let typeIns = AndesMessageTypeFactory.provide(type)
         let hierarchyIns = AndesMessageHierarchyFactory.provide(hierarchy, forType: typeIns)
-        return AndesMessageViewConfig(backgroundColor: hierarchyIns.backgroundColor, pipeColor: hierarchyIns.pipeColor, textColor: hierarchyIns.textColor, titleText: title, bodyText: body, icon: typeIns.icon, iconBackgroundColor: hierarchyIns.accentColor, isDismissable: isDismissable, dismissIconColor: hierarchyIns.textColor)
+
+        var config = AndesMessageViewConfig(backgroundColor: hierarchyIns.backgroundColor, pipeColor: hierarchyIns.pipeColor, textColor: hierarchyIns.textColor, titleText: title, bodyText: body, icon: typeIns.icon, iconBackgroundColor: hierarchyIns.accentColor, isDismissable: isDismissable, dismissIconColor: hierarchyIns.textColor)
+
+        if let pText = primaryActionText {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift]: Pongámosle `primaryText` mejor, queda más claro para qué es la constante

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift [hunk 2]
```diff
@@ -7,9 +7,20 @@
...
+            config.primaryActionConfig = AndesButtonViewConfigFactory().provide(hierarchy: hierarchyIns.primaryButtonHierarchy, size: AndesButtonSizeMedium(), text: pText, icon: nil)
+        }
+
+        if let sText = secondaryActionText {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift]: idem, `secondaryText`

File: LibraryComponents/Classes/AndesMessage/CustomButtonHierarchy/PrimaryButtonHierarchy.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//
+
+import Foundation
+
+struct PrimaryMessageActionButtonHierarchy: AndesButtonHierarchyProtocol {
+    var idleColor: UIColor
+
+    var pressedColor: UIColor
+
+    var disableColor: UIColor  = UIColor.AndesColor.Gray.gray100
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesMessage/CustomButtonHierarchy/PrimaryButtonHierarchy.swift]: No hay un semantic color para el caso de `disabled`? Si queda fijo `gray100` no estoy seguro si es darkMode friendly

File: Example/AndesUI/Messages/Views/MessageObjCViewController.m [hunk 4]
```diff
@@ -43,4 +44,13 @@ - (void)setupStyles {
     [[self.successQuiet setType:AndesMessageTypeSuccess] setHierarchy:AndesMessageHierarchyQuiet];
 }
 
+- (void)setupActions {
+    [self.neutralQuiet setPrimaryAction:@"Primary" handler:^(AndesMessage* message){
+        NSLog(@"Primary tapped");
```
[the remark this answers is not in the document]
Comment (nrostanml) [reply Example/AndesUI/Messages/Views/MessageObjCViewController.m]: Esto es código que no se ejecuta en la test app, son archivos de ejemplo para objc, igual en el proximo pr voy a mejorar la test app y incluirlo.

File: LibraryComponents/Classes/AndesMessage/CustomButtonHierarchy/PrimaryButtonHierarchy.swift (hunk 3 above)
Comment (nrostanml) [reply LibraryComponents/Classes/AndesMessage/CustomButtonHierarchy/PrimaryButtonHierarchy.swift]: es irrelevante ya que el botón hoy en dia no hay forma de que este disabled, pero lo cambio a algo, eso esta copiado del button loud original.... https://github.com/mercadolibre/fury_andesui-ios/blob/0135a00afcb2aebd1ed01198bb2dfdc9102eb7b8/LibraryComponents/Classes/AndesButton/Hierarchy/AndesButtonHierarchyLoud.swift#L21

File: Example/AndesUI/Messages/Views/MessageViewController.swift [hunk 5]
```diff
@@ -12,43 +12,50 @@ class MessageViewController: UIViewController {
...
+
+        showHidden.setHierarchy(.transparent)
     }
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setupButtons()
         createPickerViews()
         messageView.onDismiss({[unowned self] message in self.isDismissing(message)})
-        messageView.setDismissable(true)
```
[the remark this answers is not in the document]
Comment (nrostanml) [reply Example/AndesUI/Messages/Views/MessageViewController.swift]: En android también esta el mismo issue y no lo vamos a cambiar por ahora, capaz cuando hagamos el breaking de los colores metemos ese fix

File: LibraryComponents/Classes/AndesMessage/View/WithActions/AndesMessageWithActionsView.swift [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    override func updateView() {
+        super.updateView()
+        primaryAction.setSize(.medium)
+        secondaryAction.setSize(.medium)
+
+        primaryAction.updateWithCustomConfig(config.primaryActionConfig!)
+
+        if let sConfig = config.secondaryActionConfig {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesMessage/View/WithActions/AndesMessageWithActionsView.swift]: awanten las guardas ñeri

**Pull request merged**